erpnext/crm/doctype/lead/lead.py

```diff
@@ -135,10 +135,17 @@ def create_address(self):
 
 		# do not create an address if no fields are available,
 		# skipping country since the system auto-sets it from system defaults
-		if not any([self.get(field) for field in address_fields if field != "country"]):
+		address = frappe.new_doc("Address")
+
+		mandatory_fields = [ df.fieldname for df in doc.meta.fields if df.reqd ]
+
+		if not all([self.get(field) for field in mandatory_fields]):
```

Comment:
Why we are checking all mandatory fields?

Address related mandatory fields should be enough? There are possibilities, we skipped other mandatory fields for some other reason.